java-checks/src/main/java/org/sonar/java/checks/spring/SpringConfigurationWithAutowiredFieldsCheck.java

```diff
@@ -78,6 +78,10 @@ private static void collectAutowiredFields(Tree tree, Map<Symbol, VariableTree>
       return;
     }
     VariableTree variable = (VariableTree) tree;
+    if (variable.initializer() != null) {
```

Comment:
As far as I remember Spring. `@Autowired` annotation by default has `required = true`. So the initializer won't make sense, because the project will fail if no `@Bean` found (or even won't compile). Maybe it will be nice to check that `required=false` there. On the other hand, it's an additional check, so maybe we can leave it as it is.